code/framework/java-server/src/main/java/io/github/ibuildthecloud/gdapi/request/parser/DefaultApiRequestParser.java

```diff
@@ -160,22 +162,76 @@ protected String parseClientIp(ApiRequest apiRequest, HttpServletRequest request
         String clientIp = request.getRemoteAddr();
 
         clientIp = getOverrideHeader(request, overrideClientIpHeader, clientIp);
-        clientIp = getOverrideHeader(request, FORWARDED_HEADER, clientIp, false);
+        clientIp = getOverrideHeader(request, FORWARDED_FOR_HEADER, clientIp, false);
 
         return clientIp;
     }
 
+    /**
+     * Constructs the request URL based off of standard headers in the request, falling back to the HttpServletRequest.getRequestURL() 
+     * if the headers aren't available. Here is the ordered list of how we'll attempt to construct the URL: 
+     *  - x-api-request-url
+     *  - x-forwarded-proto://x-forwarded-host:x-forwarded-port/HttpServletRequest.getRequestURI()
```

Comment:
So I guess the x-forward-proto is required if they want to use any of the other x-forwarded-* headers?